scalableminds/webknossos
# Make isosurfaces more robust

- don't crash front-end if isosurface request fails
- retry isosurface requests if they fail
- indicate loading state of isosurfaces also on initial request

### URL of deployed dev instance (used for testing):
- https://robustisosurfaces.webknossos.xyz

### Steps to test:
- open a dataset with segmentation
- set network speed to slow
- request an isosurface --> a loading icon should be visible as long as the request is pending
- set network setting to offline
- request an isosurface --> a loading icon should appear and the console should log details about retries
- set network setting to online --> the isosurface generation should succeed on the next retry

### Issues:
- follow up #5101

------
